StripePayments/StripePayments/Source/PaymentHandler/STPPaymentHandler.swift

```diff
@@ -1719,6 +1719,8 @@ public class STPPaymentHandler: NSObject {
                                 // This isn't great, but UIViewController is non-nil in the protocol. Maybe it's better to still call it, even if the VC isn't useful?
                                 context.authenticationContextWillDismiss?(UIViewController())
                             }
+                            // This isn't great, but UIViewController is non-nil in the protocol. Maybe it's better to still call it, even if the VC isn't useful?
+                            self.callContextWillDismissIfNeeded(context, UIViewController())
```

Comment:
1. I don't understand the need for instantiating `UIViewController()` why not just pass in nil ?
I'm also not sure why we replicated the comment?

2. Calling `callContextWillDismissIfNeeded`  here doesn't seem correct because the body of the function calls ` authenticationContextDidDismiss`, but right before calling this function's we've just executed `authenticationContextWillDismiss`. I'm not sure how we can be certain that the dismissal happens exactly between the execution of these lines.

Reply:
1. I just copied over the existing functionality of the willDismiss as it's nearly identical in API shape (passing the view controller). Passing in nil will just make callContextDidDismissIfNeeded no-op, so I don't think that's an option and passing nil back to the delegate doesn't make sense IMO. We could update the newly added protocol function to accept an optional UIViewController, but that inconsistency with the existing `authenticationContextWillDismiss` would seem awkward, so I leaned towards keeping the existing pattern. 

2. AFAIK, the dismiss has already happened once the `completionHandler` for the `ASWebAuthenticationSession` is invoked, so I believe it's valid here.

Reply:
I'm guessing we pass `UIViewController()`for the existing protocol function b/c the protocol was defined before we added this functionality. Then we got here and realized we can't pass nil unless we make a change to the public API, so we just pass in a dummy VC. Not great, but I'm not proposing to modify the public API signature.